core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -356,7 +356,8 @@ public boolean doAuthenticate(String userName, Object credential) throws UserSto
             String[] userDNPatternList = patterns.split("#");
             if (userDNPatternList.length > 0) {
                 for (String userDNPattern : userDNPatternList) {
-                    name = MessageFormat.format(userDNPattern, userName);
+                    name = MessageFormat.format(userDNPattern,
+                                                escapeUsernameSpecialCharacters(userName,true));
```

Comment:
It seems it's already fixed in the code : https://github.com/wso2/carbon4-kernel/blob/master/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java#L381